### PR TITLE
Add i18n material date control

### PR DESCRIPTION
**Pull request by edgarmueller**
This PR adds a material specific date control with support for i18n.

### CODE REVIEW

File: packages/i18n/src/index.ts [hunk 1]
```diff
@@ -1,29 +1,42 @@
...
 export const translateProps = (state, props) => {
   const label = translateLabel(getTranslations(state), props.label);
+  const localLocale = moment().locale(getLocale(state));
 
   return {
     ...props,
-    label
+    label,
+    locale: getLocale(state),
+    localLocale
```
Comment (eneufeld) [inline packages/i18n/src/index.ts]: shoudn't we explicitly name this "momentLocale" ?

Comment (edgarmueller) [reply packages/i18n/src/index.ts]: Makes sense, update on its way.

File: packages/material/src/controls/MaterialDateControl.tsx [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+      required,
+      path,
+      handleChange,
+      data,
+      momentLocale
+    } = this.props;
+    const isValid = errors.length === 0;
+    const trim = uischema.options && uischema.options.trim;
+    const controlElement = uischema as ControlElement;
+    const resolvedSchema = resolveSchema(schema, controlElement.scope);
```
Comment (eneufeld) [inline packages/material/src/controls/MaterialDateControl.tsx]: replace with `scopedSchema`

File: packages/material/src/controls/MaterialDateControl.tsx [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+    const description = resolvedSchema.description === undefined ? '' : resolvedSchema.description;
```
Comment (eneufeld) [inline packages/material/src/controls/MaterialDateControl.tsx]: replace with description from props

File: packages/vanilla/src/controls/InputControl.tsx [hunk 4]
```diff
@@ -33,6 +33,7 @@ export class InputControl extends Control<VanillaControlProps, ControlState> {
     const isValid = errors.length === 0;
     const divClassNames = 'validation' + (isValid ? ' ' + classNames.description : ' validation_error');
     const showDescription = !isDescriptionHidden(visible, description, this.state.isFocused);
+    const labelText = typeof label === 'string' ? label : label.default;
```
Comment (eneufeld) [inline packages/vanilla/src/controls/InputControl.tsx]: why not use `isPlainLabel`?

File: packages/vanilla/src/complex/TableArrayControl.tsx [hunk 5]
```diff
@@ -68,12 +68,13 @@ class TableArrayControl extends RendererComponent<TableProps, void> {
     const labelObject = createLabelDescriptionFrom(controlElement);
     const isValid = errors.length === 0;
     const divClassNames = 'validation' + (isValid ? '' : ' validation_error');
+    const labelText = typeof label === 'string' ? label : label.default;
```
Comment (eneufeld) [inline packages/vanilla/src/complex/TableArrayControl.tsx]: why not use `isPlainLabel`?

**Pull request merged**